crates/biome_js_semantic/src/semantic_model/builder.rs

```diff
@@ -222,14 +225,18 @@ impl SemanticModelBuilder {
                     reference_id: reference_index,
                 });
 
-                self.declared_at_by_range.insert(range, binding_id);
+                self.bindings_by_reference_range.insert(range, binding_id);
             }
             Write {
                 range,
-                declared_at: declaration_at,
+                binding_id,
                 scope_id,
+                ..
             } => {
-                let binding_id = self.bindings_by_range[&declaration_at];
+                // SAFETY: this binding id is guaranteed to exist because they were generated by the
+                // event extractor
+                debug_assert!(binding_id < self.bindings.len());
```

Comment:
nit: Usually with assertions, we try to leave a message that helps developers to fix the issue, a hint of what the issue is caused by.